motionplan/collision.go

```diff
@@ -59,42 +59,26 @@ func collisionSpecificationsFromProto(
 	worldState *referenceframe.WorldState,
 ) (allowedCollisions []*Collision, err error) {
 	// List of all names which may be specified for collision ignoring.
-	validGeoms := map[string]bool{}
+	// Can seed this map with worldState names which will never have duplicates
+	validGeoms := worldState.ObstacleNames()
 
-	addGeomNames := func(parentName string, geomsInFrame *referenceframe.GeometriesInFrame) error {
+	// Get names of all geometries in frame system
+	for frameName, geomsInFrame := range frameSystemGeometries {
+		if _, ok := validGeoms[frameName]; ok {
+			return nil, referenceframe.NewDuplicateGeometryNameError(frameName)
```

Comment:
[q] if `validGeoms` is "Can seed this map with worldState names which will never have duplicates", why do we do another check here?